app/code/Magento/Sales/Model/Order.php

```diff
@@ -1965,16 +1976,71 @@ public function getRelatedObjects()
      *
      * @return string
      */
-    public function getCustomerName()
+    public function getCustomerName(): string
     {
-        if ($this->getCustomerFirstname()) {
-            $customerName = $this->getCustomerFirstname() . ' ' . $this->getCustomerLastname();
-        } else {
-            $customerName = (string)__('Guest');
+        if (null === $this->getCustomerFirstname()) {
+            return (string)__('Guest');
+        }
+
+        $customerName = '';
+        if ($this->isVisibleCustomerPrefix() && strlen($this->getCustomerPrefix())) {
+            $customerName .= $this->getCustomerPrefix() . ' ';
+        }
+        $customerName .= $this->getCustomerFirstname();
+        if ($this->isVisibleCustomerMiddlename() && strlen($this->getCustomerMiddlename())) {
+            $customerName .= ' ' . $this->getCustomerMiddlename();
+        }
+        $customerName .= ' ' . $this->getCustomerLastname();
+        if ($this->isVisibleCustomerSuffix() && strlen($this->getCustomerSuffix())) {
+            $customerName .= ' ' . $this->getCustomerSuffix();
         }
+
         return $customerName;
     }
 
+    /**
+     * Is visible customer middlename
+     *
+     * @return bool
+     */
+    private function isVisibleCustomerMiddlename(): bool
```

Comment:
Could you please move the new private methods to the end of the class? So the public and protected methods would be followed by private methods.

Reply:
Moved, thank you.